test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java

```diff
@@ -174,11 +174,81 @@ public void testRandomUniqueNormalUsageAlwayMoreThanOne() {
         assertThat(randomUnique(() -> randomAlphaOfLengthBetween(1, 20), 10), hasSize(greaterThan(0)));
     }
 
+    public void testRandomSubsetOfWithVarargs() {
+        List<Integer> result = ESTestCase.randomSubsetOf(2, 1, 2, 3, 4);
+        assertEquals(2, result.size());
+        assertTrue(Arrays.asList(1, 2, 3, 4).containsAll(result));
+    }
+
+    public void testRandomSubsetOfWithVarargsAndSizeZero() {
+        List<Integer> result = ESTestCase.randomSubsetOf(0, 1, 2, 3);
+        assertEquals(0, result.size());
```

Comment:
> This should hold for all values of `2`

In particular, if we pick `2` randomly from a set including `0` then we'd cover this case at the same time.